test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java

```diff
@@ -624,6 +628,53 @@ private Response fetchMvLongs() throws IOException {
         return query(query.toString(), "columns");
     }
 
+    public void testLookupExplosion() throws IOException {
+        int sensorDataCount = 1000;
+        int lookupEntries = 100;
+        Map<?, ?> map = responseAsMap(lookupExplosion(sensorDataCount, lookupEntries));
+        assertMap(map, matchesMap().extraOk().entry("values", List.of(List.of(sensorDataCount * lookupEntries))));
+    }
+
+    public void testLookupExplosionManyMatches() throws IOException {
+        assertCircuitBreaks(
+            () -> System.err.println(EntityUtils.toString(lookupExplosion(1000, 10000).getEntity(), StandardCharsets.UTF_8))
```

Comment:
What's our limit here? I was thinking if we should reach the billions, just to test int limit, in case we hit in some way. But maybe this is not the place for it, just curious

Reply:
oh shoot. this shouldn't have a println still in it. I wasn't throwing with values a lot smaller than this.

Reply:
Like, we were ok if you used smaller numbers than these.

Reply:
Oh my! Pulling this locally these numbers aren't failing. I'm binary searching for a failure.

Reply:
I'll push the limit, but it's about `8500, 10000` in the heaps the test uses.